Lib/test/test_getopt.py

```diff
@@ -19,21 +19,34 @@ def assertError(self, *args, **kwargs):
         self.assertRaises(getopt.GetoptError, *args, **kwargs)
 
     def test_short_has_arg(self):
-        self.assertTrue(getopt.short_has_arg('a', 'a:'))
-        self.assertFalse(getopt.short_has_arg('a', 'a'))
+        self.assertIs(getopt.short_has_arg('a', 'a:'), True)
```

Comment:
Answering @gpshead's comment https://github.com/python/cpython/pull/126393/files#r1831644790:

> using assertIs for True and False checks is unusual and too specifically tied to the implementation detail that those happen to be singletons. Add comments anywhere you do this as to _why_ assertTrue or even assertEqual would be incorrect.
> 
> A nicer non-implementation detail tied way to express this that is probably what you were intending is: assertIsInstance(thing, bool) + assertTrue/False

This function returned a boolean, but now it returns a three-state value: `True`, `False` or `'?'`. The third value is arbitrary, but it is more convenient if it has a boolean value equal to true, so `None` was rejected.

`assertTrue()` could not distinguish `True` from `'?'`. `assertEqual()` could work, but it has no advantage over  `assertIs()`. 

assertIsInstance(thing, bool) + assertTrue/False requires storing the result in a variable or duplicating the function call. Thus making the code less compact and symmetric -- instead of 1 line per case (True/False/'?'/error), some cases would require 3 lines.

The current form looks the best for me, but if you insist, I'll rewrite.